### PR TITLE
fixes #513 by adding appropriate overloads

### LINKED ISSUE #513: Add Bind overloads that accepts conversion funcs
It's nice that you can pass in an ad-hoc `IBindingTypeConverter` when you call `Bind`. But writing one of those is a real pain in the ass. Especially in cases where all you need is two simple lambdas for the conversion and the conversion back.

This is a humble request to provide an overload that accepts two funcs.

### PR DESCRIPTION
**Pull request by moswald**

### CODE REVIEW

File: ReactiveUI/PropertyBinding.cs [hunk 1]
```diff
@@ -453,17 +598,112 @@ public class PropertyBinderImplementation : IPropertyBinderImplementation
...
+                TView view,
+                Expression<Func<TViewModel, TVMProp>> vmProperty,
+                Expression<Func<TView, TVProp>> viewProperty,
+                IObservable<TDontCare> signalViewUpdate,
+                Func<TVMProp, TVProp> vmToViewConverter,
+                Func<TVProp, TVMProp> viewToVmConverter)
+            where TViewModel : class
+            where TView : IViewFor
+        {
+            OutFunc<TVMProp, TVProp> vmToViewFunc = (TVMProp vmValue, out TVProp vValue) => {
```
Comment (kentcb) [inline ReactiveUI/PropertyBinding.cs]: For the avoidance of confusion, should we throw `ArgumentNullException` here if either `vmToViewConverter` or `viewToVmConverter` is `null`?

File: ReactiveUI/PropertyBinding.cs [hunk 2]
```diff
@@ -453,17 +598,112 @@ public class PropertyBinderImplementation : IPropertyBinderImplementation
...
+                TViewModel viewModel,
+                TView view,
+                Expression<Func<TViewModel, TVMProp>> vmProperty,
+                Expression<Func<TView, TVProp>> viewProperty,
+                IObservable<TDontCare> signalViewUpdate,
+                Func<TVMProp, TVProp> vmToViewConverter,
+                Func<TVProp, TVMProp> viewToVmConverter)
+            where TViewModel : class
+            where TView : IViewFor
+        {
```
Comment (kentcb) [inline ReactiveUI/PropertyBinding.cs]: For the avoidance of confusion, should we throw `ArgumentNullException` here if either `vmToViewConverter` or `viewToVmConverter` is `null`?

Comment (moswald) [inline ReactiveUI/PropertyBinding.cs]: That's fine with me.

Comment (kentcb) [inline ReactiveUI/PropertyBinding.cs]: @moswald Thanks - I see you've made these changes, but they appear to be in a different branch...? Can you include them in this PR?

Comment (moswald) [inline ReactiveUI/PropertyBinding.cs]: lol. Whoops.

**Comment by kentcb** [comment]
Great - thanks @moswald! This will be super useful.

**Pull request merged**